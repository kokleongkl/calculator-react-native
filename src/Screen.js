import React from "react";
import { View, Text } from "react-native";
import { NUMBER_BUTTONS, SYMBOL_BUTTONS } from "./constants";
import { styles } from "./styles";
import Button from "../src/Button";
import { connect } from "react-redux";
import { handleOnClick } from "./reducer";

const Screen = (props) => (
  <View style={styles.rootContainer}>
    <View style={styles.displayOutput}>
      <Text style={styles.textDefault}>{props.output}</Text>
    </View>
    <View style={styles.containerRow}>
      <View style={styles.flexNumbers}>
        {NUMBER_BUTTONS.map((row, index) => (
          <View key={index} style={styles.containerRow}>
            {row.map((col, index) => (
              <Button
                key={index}
                onPress={() => handleOnClick(col, props)}
                text={col}
              />
            ))}
          </View>
        ))}
      </View>
      <View style={styles.rootContainer}>
        {SYMBOL_BUTTONS.map((row, index) => (
          <Button
            style={{ marginRight: 10 }}
            key={index}
            onPress={() => handleOnClick(row, props)}
            text={row}
          />
        ))}
      </View>
    </View>
  </View>
);

const mapStateToProps = (state) => ({ output: state.dispatchReducer.output });
const mapDispatchToProps = (dispatch) => {
  const {
    clear,
    equals,
    replace,
    concat,
    concatwithout,
  } = require("./actions");
  return {
    clear: () => dispatch(clear()),
    equals: () => dispatch(equals()),
    replace: (value) => dispatch(replace(value)),
    concat: (value) => dispatch(concat(value)),
    concatwithout: (value) => dispatch(concatwithout(value)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Screen);
