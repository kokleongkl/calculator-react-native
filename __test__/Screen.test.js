import React from "react";
import Screen from "../src/Screen";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({ dispatchReducer: { output: "0" } });

describe("Screen Component", () => {
  describe("Screen Rendered", () => {
    it("should match to snapshot", () => {
      const testcomponent = renderer
        .create(
          <Provider store={store}>
            <Screen />
          </Provider>
        )
        .toJSON();
      expect(testcomponent).toMatchSnapshot();
    });
  });
});
