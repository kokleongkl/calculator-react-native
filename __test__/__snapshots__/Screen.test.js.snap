// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Component Screen Rendered should match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 60,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              7
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              8
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              9
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              5
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              6
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              2
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              3
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              C
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              0
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "margin": 4,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              .
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "margin": 4,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            /
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "margin": 4,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            *
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "margin": 4,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "margin": 4,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            -
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "margin": 4,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ddd",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
